cpp/src/arrow/compute/exec/asof_join_node_test.cc

```diff
@@ -633,6 +633,11 @@ struct BasicTest {
 
 using AsofJoinBasicParams = std::tuple<std::function<void(BasicTest&)>, std::string>;
 
+void PrintTo(const AsofJoinBasicParams& x, ::std::ostream* os) {
+  *os << "AsofJoinBasicParams: " << std::get<0>(x).target_type().name() << " - "
+      << std::get<1>(x);
```

Comment:
```suggestion
  *os << "AsofJoinBasicParams: " << std::get<1>(x);
```

I think `target_type().name()` is unlikely to add any useful information beyond `std::get<1>(x)`.  CC @rtpsw for a second opinion.

Reply:
@westonpace is correct. As long as the [test names](https://github.com/apache/arrow/blob/40ec95646962cccdcd62032c80e8506d4c275bc6/cpp/src/arrow/compute/exec/asof_join_node_test.cc#L915-L920) are unique, the test name inserted with `std::get<1>(x)` will be sufficient.

Reply:
thanks